src/codestyle.sh

```diff
@@ -96,9 +98,15 @@ function codestyle_main()
     end_line=$(wc -l < "$FLIST")
   fi
 
-  # If --start-line and --end-line options are being used, assign a temporary file
-  # to FLIST with the line interval and SPDX License Identifier
-  if [[ -n "$range_option_flag" ]]; then
+  # For --function option, find the start line of the function definition
+  # and assign it to start_line.
+  if [[ -n "$function" ]]; then
+    start_line=$(awk "/[a-zA-Z]+[[:space:]]+${function}[[:space:]]*\\(.*[^;]$/ {print FNR}" "$FLIST")
+  fi
```

Comment:
Check the `src/lib/kw_string` to see if we already have something like that; if not, this is a great candidate to become a generic function in the `kw_string` file.